### ISSUE TITLE
If you build the library and the executable at the same time, embed-resource will fail

### ISSUE THREAD

**Issue by byeblack** [nabijaczleweli/rust-embed-resource#42]
embed-resource fails when lib.rs exists in the project, or [lib] is in Cargo.toml

Windows 10 
Rust Version is 1.59.0
embed-resource Version is 1.6.5

**Comment by noe132** [comment]
Happened to me as well, when lib.rs presents, the executable built was the same as if embed-resource wasn't used, and no error messages.

**Comment by nabijaczleweli** [comment]
Do you have a log or an example that reproduces this? I don't really see a reason this shouldn't work, and AFAIK it has done so far?

**Comment by noe132** [comment]
@nabijaczleweli 
https://github.com/noe132/rust-embed-resource-lib-bug
run `cargo build -r` the exe would have no icon
delete `src/lib.rs` and run `cargo build -r` again the exe would have icon as expected.

![Video_2022-02-28_112132](https://user-images.githubusercontent.com/2271900/155918916-4a49e114-5866-40e3-b4df-9515128d5aa2.gif)

on windows 10 and rust 1.59.0

**Comment by nabijaczleweli** [comment]
uuh
```
T:\>git diff withlib nolib -U0
diff --git a/withlib b/nolib
index 095dcfb..26f749f 100644
--- a/withlib
+++ b/nolib
@@ -1 +1 @@
-T:\rust-embed-resource-lib-bug>cargo build -v > withlib
+T:\rust-embed-resource-lib-bug>cargo build -v > nolib
@@ -7 +6,0 @@ T:\rust-embed-resource-lib-bug>cargo build -v > withlib
-     Running `rustc --crate-name rust_demo --edition=2018 src\lib.rs --error-format=json --json=diagnostic-rendered-ansi,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=d71b703fa29c9940 -C extra-filename=-d71b703fa29c9940 --out-dir T:\rust-embed-resource-lib-bug\target\debug\deps -C incremental=T:\rust-embed-resource-lib-bug\target\debug\incremental -L dependency=T:\rust-embed-resource-lib-bug\target\debug\deps -L native=T:\rust-embed-resource-lib-bug\target\debug\build\rust-demo-433fe507bd489039\out -l dylib=icon`
@@ -9 +7,0 @@ T:\rust-embed-resource-lib-bug>cargo build -v > withlib
-     --extern rust_demo=T:\rust-embed-resource-lib-bug\target\debug\deps\librust_demo-d71b703fa29c9940.rlib
@@ -10,0 +9 @@ T:\rust-embed-resource-lib-bug>cargo build -v > withlib
+     -l dylib=icon
```

yeah, confirmed. i thought this would carry over transitively but it .. doesn't (or, maybe, it used to, but doesn't anymore?) – only the library is linked with the icon. hm.

**Comment by nabijaczleweli** [comment]
Can you try with latest `master` branch (at least e895286a9ac88c04d113a35dcc09cf615e32e405)? Your PoC worked for me.

**Comment by byeblack** [comment]
> Can you try with latest `master` branch (at least [e895286](https://github.com/nabijaczleweli/rust-embed-resource/commit/e895286a9ac88c04d113a35dcc09cf615e32e405))? Your PoC worked for me, but I'd be much happier with a confirmation from a real use-case (do note that you need rustc >= 1.50.0, there's no way to fix this in earlier toolchains).

Thanks a lot, it worked successfully in my project

**Comment by noe132** [comment]
Works for me too on `master` 👍

**Comment by nabijaczleweli** [comment]
Great! Released in [`v1.6.6`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.6.6).
